src/tiled/abstracttilefilltool.cpp

```diff
@@ -76,8 +76,8 @@ void AbstractTileFillTool::deactivate(MapScene *scene)
 
 void AbstractTileFillTool::mousePressed(QGraphicsSceneMouseEvent *event)
 {
-    if (event->button() == Qt::RightButton && event->modifiers() == Qt::NoModifier) {
```

Comment:
Ah, wait a sec, this "no modifier" check is to avoid conflicting with the "select top-most tile layer with a tile at this location" behavior that is implemented in `AbstractTileTool::mousePressed`. So, this should check that the Ctrl modifier isn't held.